### PR TITLE
sarahpeng/phs-26-spotlights-page-basic-screens

### PR DESCRIPTION
**Pull request by sarahhpeng**
# ✨ New in this PR ✨

## One liner: what did you do? 🛠
Created the spotlight page home page that takes you to a spotlight page, which then takes you to display pages. 

## Coverage 🙆‍♀️
Created the spotlight home page
- first, created a query called fetchSpotlightTours, filtering through the tours to only get the tours that were spotlights (spotlight value is true)
- then, worked on the front end, including the nav bar, the title, the brief description, and then mapping each spotlight to display the name and description underneath a rounded box where the image would go (andrei is working on this)
- each spotlight is a link that takes you to a new page, the spotlight page

Created the spotlight page
- after you go from the home page to a spotlight page. again, i created queries that fetched tours, taking in the spotlight ID, and then a query called fetchDisplayfromSpotlight that fetched all the displays corresponding with that spotlight ID
- then, worked on the front end again, including the nav bar, and then the spotlight name and description
- then, mapped displays to display all the displays as green buttons that would wrap as the content overflowed from the screen. each button is a clickable link that takes you to the display page (i used  <Link href={`/spotlightPage/${spotlight.id}/${display.id}?spotlightId=${spotlight.id}`} > for the href so that spotlight.id would still be kept as you moved into the display page)

Created the display page
- once you click on one of the buttons, you go into a display page. i created a query called fetchDisplayFromId that takes in the display id and gets the display row information of that id. then i set that to currentDisplayData
- i also created a const called otherDisplays that takes in the spotlight id, and filters through so that the current display id is not included in the list. this is for later on when we map and display the buttons that lead to 'More in this spotlight...' displays
- on the front end, similarly, displayed the display title and description. NOTE: the description under the title is missing, and has to be added to the supabase table.
- then mapped the otherDisplays, wrapped with a Link, so that the user can click in to the other displays. these buttons also scroll horizontally and overflow on the right side of the screen!

TLDR of what still needs to be done: 
- images! 
- add spotlight page to nav bar
- display small description
- define what related spotlights are

## How can the reviewer test your code? Where is this sprint located in the codebase? 👩‍💻
- localhost:3000/spotlightPage. then click around.

## REQUIRED: Link a short video demo of the feature you have added. 👩‍💻

https://github.com/calblueprint/phs/assets/137237272/683f88ce-7cfe-40d0-953a-2d5affc53a88

## Any bugs you encountered or still having trouble with? 🐛
- Encountered lots of bugs, but none right now... i think

## Resources 📔
- Andrei and Chris

_REQUIREMENTS in this section: Link all relevant online resources and related PRs. Write down any new dependencies or packages and link to their documentation. If you ran into bugs, link any stack overflow or online resources that helped you unblock yourself_

## ❗IF this sprint was marked as FRONTEND on linear, cc Claire Wang

_REMEMBER: once PR is made, also request a PR review Andrei, CC'ing is NOT enough!_
🏚 cc: @andreisito
cc: @cluurie

### CODE REVIEW

File: src/app/spotlightPage/[spotlightId]/[displayId]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable */
```
Comment (oahnh) [inline src/app/spotlightPage/[spotlightId]/[displayId]/page.tsx]: We shouldnt disable this 🥲 lmk if u need help debugging the eslint stuff

File: src/app/spotlightPage/[spotlightId]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/* eslint-disable */
```
Comment (miha-bhaskaran) [inline src/app/spotlightPage/[spotlightId]/page.tsx]: same as what noah mentioned above!

File: src/app/spotlightPage/[spotlightId]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        alt="placeholder"
+        height={145}
+      />
+      <h1 className="text-green-700 font-lato text-base font-normal pl-4 pt-[31px]"> CATEGORY TWO</h1>
+      <h1 className="text-[#333333] text-3xl text-14 font-bold pl-4 pt-[8px]">
+        {spotlight.name}
+      </h1>
+      <p className="text-[#333333] p-4">{spotlight.description}</p>
+      <h1 className="text-black font-lato text-16 font-medium pl-4">In this spotlight...</h1>
+      {/* TODO: Add display ids */}
```
Comment (miha-bhaskaran) [inline src/app/spotlightPage/[spotlightId]/page.tsx]: remove this

**Review by miha-bhaskaran — DISMISSED** [review]
Just make sure eslint stuff are not disabled and fully resolved. Also make sure that ur error catching message is reflecting the actual function - if its fetching, erroring, creating, updating

**Review by cluurie — COMMENTED** [review]
hellooo one thing visual-wise: take note of margins (keeping them consistent for all components on the page) AND spacing (if u click/hover over objects in the figma in dev mode, the pink indication will tell u the spacing in px between two objects)

File: src/app/spotlightPage/[spotlightId]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          {relatedSpolights.map(
+            otherSpotlight => (
+                <li className="pl-[18px] w-[162px] overflow-x-auto" key={otherSpotlight.id}>
+                  <Link
+                    href={`/spotlightPage/${spotlight.id}`}>
+                    <div className="bg-[#386131] h-[169px] rounded-2xl p-[18px] flex flex-col" />
+                    <h4 className="text-black font-Lato text-20 font-bold mt-2">
+                          {otherSpotlight.name}
+                        </h4>
+                        <h2 className="text-gray font-Lato text-sm font-normal">
```
Comment (andreisito) [inline src/app/spotlightPage/[spotlightId]/page.tsx:97]: for hifi sprint:  change this to text-shadow, this text is a little light.

File: src/app/spotlightPage/[spotlightId]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                          {otherSpotlight.description}
+                        </h2>
+                  </Link>
```
Comment (andreisito) [inline src/app/spotlightPage/[spotlightId]/page.tsx:100]: HiFi: For this block of code, lets put it into its own component and import here so we can have better readability -> limit # of lines in this file

File: src/app/spotlightPage/[spotlightId]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  
+
+  return (
+    <div className="bg-[#ebf0e4]">
+      <NavBar />
+      <img
+        src="https://images.unsplash.com/photo-1615812214207-34e3be6812df?auto=format&fit=crop&q=80&w=2940&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
+        alt="placeholder"
+        height={145}
+      />
```
Comment (andreisito) [inline src/app/spotlightPage/[spotlightId]/page.tsx:63]: in hifi, turn this into a Image NextJS tag

File: src/app/spotlightPage/[spotlightId]/[displayId]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    title: "N/A",
+    updated_at: "N/A",
+  });
+
+  const [otherDisplays, setOtherDisplays] = useState<DisplayRow[]>([]);
+
+  useEffect(() => {
+    /**
+     * @returns a display 
+     */
```
Comment (andreisito) [inline src/app/spotlightPage/[spotlightId]/[displayId]/page.tsx:35]: for fetch data, add a more descriptive comment on what is happening within the function (2-3 sentences on average)

File: src/app/spotlightPage/[spotlightId]/page.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+  const [displays, setDisplays] = useState<DisplayRow[]>([])
+
+  const [relatedSpolights, setRelatedSpotlight] = useState<TourRow[]>([])
+
+  useEffect(() => {
+    /**
+     * @returns data from tour table and display table
+     */
+    async function fetchData() {
```
Comment (andreisito) [inline src/app/spotlightPage/[spotlightId]/page.tsx:38]: more descriptive comments on internal fetchData logic.  ex: explain variables we are creating/setting and what data we are retrieving from

File: src/app/spotlightPage/page.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+import Link from 'next/link';
+import React, { useEffect, useState } from 'react';
+import Image from 'next/image';
+
+import { fetchSpotlightTours } from '../../supabase/tours/queries';
+import NavBar from '../../components/userComponents/navBar/navBar';
+import { TourRow} from '../../types/types';
+
+/**
+ * @returns spotlights from tours table
```
Comment (andreisito) [inline src/app/spotlightPage/page.tsx:12]: mention how this is the home page and that it dynamically loads the following parts of hte spotlights flow.  Add desc of the general spotlight flow here

File: src/app/spotlightPage/page.tsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+                      </div>
+                      
+                    </div>
+                    <h4 className="text-black font-Lato text-20 font-bold pt-[14px]">
+                          {spotlight.name}
+                        </h4>
+                        <h2 className="text-black font-Lato text-sm font-normal pt-[4px]">
+                          {spotlight.description}
+                        </h2>
+                  </Link>
```
Comment (andreisito) [inline src/app/spotlightPage/page.tsx:74]: For maintenance and readability -> turn this into its own component and import back to this file

**Review by andreisito — APPROVED** [review]
nice.  Will transition styles to use global styling in followup sprints + will move some of the queries into Database/SQL queries in the future

**Review by andreisito — APPROVED** [review]

**Comment by cluurie** [comment]
**For all pages**
- [ ] background color is #F6F6F6, change to #FFFDF7
- [ ] adjust spacing between everything (text, images, buttons, icons)
- [ ] check all text font weights, styles, and sizes

**Our Wildlife Spotlights Page**
- [ ] add top left back button (see figma file for updated design) --> note the spacing between button and header text (12px) and spacing between navbar and back button (16px)
- [ ] decrease media image button rounding to 8px

**Dove Flight Enclosures + Test Display Pages**
- [ ] change "Category Two" text color to #386131
- [ ] decrease spotlight button rounding to 8px
- [ ] change spotlight buttons color to 7CA24E
- [ ] change spotlight button dimensions to width 168px x height 74px
- [ ] decrease gap between rows of spotlight buttons to 12px
- [ ] check spacing between buttons in horizontal scroll 
- [ ] note that spotlight buttons are a different dimension than the horizontal scroll buttons

_yippeee thank u 👯 _
